### PR TITLE
feat(coupons): coupon detail api check valid promo

### PR DESCRIPTION
**Pull request by StaberindeZA**
## Because

- The coupon details api endpoint was only checking if a coupon is valid
  and was not also checking the promotion code.

## This pull request

- Verify's that the coupon and promotion code are valid. If either is
  invalid, return the appropriate information so that the UI can display
  the correct error message.

## Issue that this pull request solves

Closes: #11524

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [x] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/payments/stripe.ts [hunk 1]
```diff
@@ -745,6 +745,88 @@ export class StripeHelper {
...
+    }
+
+    if (
+      verifyCoupon.maximallyRedeemed ||
+      verifyPromotionCode.maximallyRedeemed
+    ) {
+      returnValue.maximallyRedeemed = true;
+    }
+
+    return returnValue;
```
Comment (bbangert) [inline packages/fxa-auth-server/lib/payments/stripe.ts:827]: Maybe just...
```typescript
return {
  expired: verifyCoupon.expired || verifyPromotionCode.expired,
  valid: verifyCoupon.valid && verifyPromotionCode.valid,
  maximallyRedeemed: verifyCoupon.maximallyRedeemed || verifyPromotionCode.maximallyRedeemed
}
```

Then you could also drop its initialization.

**Review by bbangert — CHANGES_REQUESTED** [review]

File: packages/fxa-auth-server/lib/payments/stripe.ts (hunk 1 above)
Comment (StaberindeZA) [reply packages/fxa-auth-server/lib/payments/stripe.ts:827]: I'm also not super happy with my current implementation, but I mainly did it this way to keep the API behavior the same.

Currently we only return `expired` and `maximallyRedeemed` when they have a value of `true`, else the properties aren't included in the return object.

Is there a better way to achieve this, or should we change the logic to return true/false instread of true/undefined.

Comment (bbangert) [reply packages/fxa-auth-server/lib/payments/stripe.ts:827]: Is there any harm in always returning them even when they're false?

Comment (StaberindeZA) [reply packages/fxa-auth-server/lib/payments/stripe.ts:827]: Nope, I don't think so. Just changing a bunch of tests, but that's about it.

Comment (bbangert) [reply packages/fxa-auth-server/lib/payments/stripe.ts:827]: I'm fine leaving it or changing it, if its a whole bunch of tests, maybe we file a bug to trim it down a bit later with that simplification.

**Review by bbangert — APPROVED** [review]

**Pull request merged**